src/resmom/linux/mom_start.c

```diff
@@ -2081,6 +2081,38 @@ set_job(job *pjob, struct startjob_rtn *sjr)
 					 */
 			sjr->sj_reservation = 0;
 		}
+		
+		/*
+		 * If we have a valid reservation id, save it off in case
+		 * the ALPS reservation information isn't part of the job struct
+		 * (e.g. job got killed before ALPS reservation ID was noted)
+		 * Then we can use this to cancel the ALPS reservation so we
+		 * don't leave orphaned ALPS reservations around.
```

Comment:
this comment is also too verbose. We can just say "Save the ALPS reservation ID so that it can be used to delete the reservation in case of race conditions where it doesn't get recorded in the job structure."